File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPArrowsInBody.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPArrowsInBody implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "arrows_in_body";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 0;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getArrowsInBody();
    }

    @Override
    public void restoreProperty(Player player, Integer arrows) {
        player.setArrowsInBody(arrows);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPArrowsInBody.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPArrowsInBody extends PlayerStateProperty[Integer]:
  override def getConfigName: String = "arrows_in_body"

  override def getDefaultSerializedProperty: Integer = 0

  override def getSerializedPropertyFromPlayer(player: Player): Integer =
    player.getArrowsInBody

  override def restoreProperty(player: Player, arrows: Integer): Unit =
    player.setArrowsInBody(arrows)